lib/lz4.h

```diff
@@ -50,9 +50,17 @@ extern "C" {
 #define LZ4_VERSION_MAJOR    1    /* for breaking interface changes  */
 #define LZ4_VERSION_MINOR    7    /* for new (non-breaking) interface capabilities */
 #define LZ4_VERSION_RELEASE  2    /* for tweaks, bug-fixes, or development */
+
 #define LZ4_VERSION_NUMBER (LZ4_VERSION_MAJOR *100*100 + LZ4_VERSION_MINOR *100 + LZ4_VERSION_RELEASE)
 int LZ4_versionNumber (void);
 
+#define QUOTE(str) #str
```

Comment:
OK, so this is the `zstdcli.c` construction re-used as-is.

It works, although there is a subtle difference here :
`zstdcli.c` declarations remain "private", only for its module.
`lz4.h` declarations will be included. 
As a consequence, any public macro / type / function will pollute the naming space of importing project.

To reduce risks of naming collisions, it is important to prefix any imported definition with `LZ4_`.
Here, it applies to `QUOTE` and to `EXPAND_AND_QUOTE`  (you may select a shorter name for this one if you feel it helps readability).